python/paddle/fluid/optimizer.py

```diff
@@ -76,24 +87,43 @@ def get_opti_var_name_list(self):
         return self._opti_name_list
 
     def _create_global_learning_rate(self):
-        lr = self._global_learning_rate()
-
-        if isinstance(lr, framework.Variable):
-            return
+        if imperative_base.enabled():
+            # create learning rate Variable
+            if isinstance(self._learning_rate, float):
+                self._learning_rate_map[framework.default_main_program(
+                )] = layers.create_global_var(
```

Comment:
can this be created many times?

Reply:
no, this should only be called once Optimizer.__init__

Reply:
Done